### PR TITLE
update to PHP 7.4 and codeception/verify 2.2

### PR DESCRIPTION
**Pull request by developedsoftware**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | yes - cannot be run on php 5
| Tests pass?   | yes
| Fixed issues  | none

This pull request is to bump codeception/verify up to 2.2 which requires PHP 7 as a minimum.

### CODE REVIEW

**Comment by developedsoftware** [comment]
Apologies for all these tests failing I am unable to test locally. Can anybody advise what codeception-local.php should contain? I am a bit confused regarding the folder structures...

```
==== Redirecting to Composer-installed version in vendor/codeception. You can skip this using --no-redirect ====

Included Configuration: common
Building Actor classes for suites: unit

In Yii2.php line 265:
                                                                               
  Yii2 module is not configured!                                               
                                                                               
  The application config file does not exist: /Users/luke.english/NetBeansPro  
  jects/yii/yii2-app-advanced/common/config/codeception-local.php              
                                                                               

build

Done.
```

File: composer.json [hunk 1]
```diff
@@ -14,10 +14,10 @@
     },
     "minimum-stability": "dev",
     "require": {
-        "php": ">=5.6.0",
+        "php": ">=7.4.0",
         "yiisoft/yii2": "~2.0.14",
         "yiisoft/yii2-bootstrap4": "~2.0.0",
-        "yiisoft/yii2-swiftmailer": "~2.0.0 || ~2.1.0"
+        "yiisoft/yii2-symfonymailer": "dev-master"
```
Comment (samdark) [inline composer.json:20]: Should be ~2.0.3.

File: environments/dev/common/config/main-local.php [hunk 2]
```diff
@@ -10,11 +10,7 @@
             'charset' => 'utf8',
         ],
         'mailer' => [
-            'class' => 'yii\swiftmailer\Mailer',
-            'viewPath' => '@common/mail',
```
Comment (samdark) [inline environments/dev/common/config/main-local.php:14]: This should stay.

File: environments/prod/common/config/main-local.php [hunk 3]
```diff
@@ -10,8 +10,8 @@
             'charset' => 'utf8',
         ],
         'mailer' => [
-            'class' => 'yii\swiftmailer\Mailer',
-            'viewPath' => '@common/mail',
```
Comment (samdark) [inline environments/prod/common/config/main-local.php:14]: This should stay.

File: environments/prod/common/config/main-local.php [hunk 4]
```diff
@@ -10,8 +10,8 @@
...
+            'class' => 'yii\symfonymailer\Mailer',
+            'useFileTransport' => true,
```
Comment (samdark) [inline environments/prod/common/config/main-local.php:14]: Production should not write to files.

**Comment by samdark** [comment]
No worries about failing tests. As long as these are finally OK, feel free to do as many intermediate commits as you need. Can't help with `codeception-local.php` :(

File: environments/prod/common/config/main-local.php (hunk 3 above)
Comment (developedsoftware) [reply environments/prod/common/config/main-local.php:14]: I thought swiftmailer was abandoned ?

File: environments/prod/common/config/main-local.php (hunk 4 above)
Comment (developedsoftware) [reply environments/prod/common/config/main-local.php:14]: You mean useFileTransport should be set to false ?

**Comment by themikem** [comment]
> Apologies for all these tests failing I am unable to test locally. Can anybody advise what codeception-local.php should contain? I am a bit confused regarding the folder structures...
> 
> ```
> ==== Redirecting to Composer-installed version in vendor/codeception. You can skip this using --no-redirect ====
> 
> Included Configuration: common
> Building Actor classes for suites: unit
> 
> In Yii2.php line 265:
>                                                                                
>   Yii2 module is not configured!                                               
>                                                                                
>   The application config file does not exist: /Users/luke.english/NetBeansPro  
>   jects/yii/yii2-app-advanced/common/config/codeception-local.php              
>                                                                                
> 
> build
> 
> Done.
> ```

This is just loaded by the module here:
https://github.com/Codeception/module-yii2/blob/14269d059b8eaedf3d414a673907bd874cd4ed04/src/Codeception/Module/Yii2.php#L288

From the yaml file here:
https://github.com/yiisoft/yii2-app-advanced/blob/1f4b6b7eafd0087fd79b991b2639998e1936e68a/common/codeception.yml#L15

It is just rolling up the main & test config files and providing the cookie key for compatibility it appears.

I think the error you're seeing can be fixed by doing a ./init and then choosing the dev environment. That should copy the correct codeception-local.php file into place.

**Comment by developedsoftware** [comment]
> > Apologies for all these tests failing I am unable to test locally. Can anybody advise what codeception-local.php should contain? I am a bit confused regarding the folder structures...
> > ```
> > ==== Redirecting to Composer-installed version in vendor/codeception. You can skip this using --no-redirect ====
> > 
> > Included Configuration: common
> > Building Actor classes for suites: unit
> > 
> > In Yii2.php line 265:
> >                                                                                
> >   Yii2 module is not configured!                                               
> >                                                                                
> >   The application config file does not exist: /Users/luke.english/NetBeansPro  
> >   jects/yii/yii2-app-advanced/common/config/codeception-local.php              
> >                                                                                
> > 
> > build
> > 
> > Done.
> > ```
> 
> This is just loaded by the module here: https://github.com/Codeception/module-yii2/blob/14269d059b8eaedf3d414a673907bd874cd4ed04/src/Codeception/Module/Yii2.php#L288
> 
> From the yaml file here:
> 
> https://github.com/yiisoft/yii2-app-advanced/blob/1f4b6b7eafd0087fd79b991b2639998e1936e68a/common/codeception.yml#L15
> 
> It is just rolling up the main & test config files and providing the cookie key for compatibility it appears.
> 
> I think the error you're seeing can be fixed by doing a ./init and then choosing the dev environment. That should copy the correct codeception-local.php file into place.

Awesome thank you

**Review by samdark — APPROVED** [review]

**Pull request merged**